spylib/utils/session_token.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+from typing import Any, Dict, Optional
+from urllib.parse import urlparse
+
+from jwt import decode
+from pydantic import root_validator
+from pydantic.main import BaseModel
+from pydantic.networks import HttpUrl
+
+from .domain import store_domain
+
+REQUIRED_FIELDS = ['iss', 'dest', 'sub', 'jti', 'sid']
+ALGORITHM = 'HS256'
+PREFIX = 'Bearer '
+
+
+class ValidationError(Exception):
+    pass
+
+
+class InvalidIssuerError(ValidationError):
+    pass
+
+
+class MismatchedHostError(ValidationError):
+    pass
+
+
+class TokenAuthenticationError(ValidationError):
+    pass
```

Comment:
As discussed let's avoid `ValidationError`.
Also to make them easier to find, it might be good practice to move these to a `session_token_exceptions.py` ... or not, I don't know.

```suggestion
class TokenValidationError(Exception):
    pass


class InvalidIssuerError(TokenValidationError):
    pass


class MismatchedHostError(TokenValidationError):
    pass


class TokenAuthenticationError(TokenValidationError):
    pass
```

Reply:
If you don't mind, I wouldn't mind leaving them bundled with the object for now as it's the only thing throwing these errors. If we use them in the future in a different object, could we move them then?